syntax = "proto3";

service NumberService {
  rpc Next (NumberRequest) returns (NumberResponse);
  rpc Range (RangeRequest) returns (stream RangeResponse);
  rpc Sum (stream SumRequest) returns (SumResponse);
}

message NumberRequest {  
}

message NumberResponse {
  uint64 value = 1;
}

message RangeRequest {
  uint64 min = 1;
  uint64 max = 2;
}

message RangeResponse {
  uint64 value = 1;
}

message SumRequest {
  uint64 value = 1;
}

message SumResponse {
  uint64 value = 1;
}